keras_cv/models/adain.py

```diff
@@ -0,0 +1,228 @@
+            layers.Conv2D(filters=256, **config),
+            layers.UpSampling2D(),
+            layers.Conv2D(filters=128, **config),
+            layers.Conv2D(filters=128, **config),
+            layers.UpSampling2D(),
+            layers.Conv2D(filters=64, **config),
+            layers.Conv2D(
+                filters=3,
+                kernel_size=3,
+                strides=1,
+                padding="same",
+                activation="sigmoid",
+            ),
+        ]
+    )
+    return decoder    
+
+
+def get_loss_net(image_size=(None, None)):
+    vgg19 = keras.applications.VGG19(
+        include_top=False, weights="imagenet", input_shape=(*image_size, 3)
+    )
+    vgg19.trainable = False
+    layer_names = ["block1_conv1", "block2_conv1", "block3_conv1", "block4_conv1"]
+    outputs = [vgg19.get_layer(name).output for name in layer_names]
+    mini_vgg19 = keras.Model(vgg19.input, outputs)
+
+    inputs = layers.Input([*IMAGE_SIZE, 3])
+    mini_vgg19_out = mini_vgg19(inputs)
+    return keras.Model(inputs, mini_vgg19_out, name="loss_net")
+
+
+def get_mean_std(x):
+    """Computes the mean and standard deviation of a tensor."""
+    axes = [1, 2]
+    mean, variance = tf.nn.moments(x, axes=axes, keepdims=True)
+    standard_deviation = tf.sqrt(variance + 1e-5)
+    return mean, standard_deviation
+
+
+class AdaIN(layers.Layer):
+    def __init__(self, name='AdaIN', **kwargs):
+        super(AdaIN, self).__init__(name=name, **kwargs)
+
+    def call(self, inputs):
+        """Computes the AdaIn feature map.
+
+        Args:
+            style: The style feature map.
+            content: The content feature map.
+
+        Returns:
+            The AdaIN feature map.
+        """
+        style, content = inputs
+        content_mean, content_std = get_mean_std(content)
+        style_mean, style_std = get_mean_std(style)
+        t = style_std * (content - content_mean) / content_std + style_mean
+        return t
+
+
+class AdaINModel(keras.Model):
+    def __init__(self, image_size=(None, None), style_weight=4.0, **kwargs):
+        super().__init__(**kwargs)
+        self.image_size = image_size
+        self.encoder = get_mini_vgg19_encoder(image_size)
+        self.decoder = get_adain_decoder()
+        self.adain_layer = AdaIN()
+        self.loss_net = get_loss_net(image_size)
+        self.style_weight = style_weight
+
+    def produce_functional_model():
+        style_input = layers.Input([*self.image_size, 3])
+        style_encoded = self.encoder(style_input)
+        content_input = layers.Input([*self.image_size, 3])
+        content_encoded = self.encoder(content_input)
+
+        adain_feature_map = self.adain_layer((style_encoded, content_encoded))
+        content_decoded = self.decoder(adain_feature_map)
+        
+        return keras.Model([style_input, content_input], content_decoded, name="adain_style_transfer")
+
+
+    def compile(self, optimizer, loss_fn):
```

Comment:
you can drop the custom method here and just rely on `self.optimizer` and `self.compiled_loss` instead of `self.loss_fn`